garak/translators/remote.py

```diff
@@ -0,0 +1,133 @@
+        "hi", "ko", "et", "sl", "bg",
+        "uk", "hr", "ar", "vi", "tr",
+        "id", "cs", "en"
+    ]
+    # fmt: on
+
+    # avoid attempt to pickle the client attribute
+    def __getstate__(self) -> object:
+        self._clear_translator()
+        return dict(self.__dict__)
+
+    # restore the client attribute
+    def __setstate__(self, d) -> object:
+        self.__dict__.update(d)
+        self._load_translator()
+
+    def _clear_translator(self):
+        self.client = None
+
+    def _load_translator(self):
+        if not (
+            self.source_lang in self.bcp47_support
+            and self.target_lang in self.bcp47_support
+        ):
+            raise BadGeneratorException(
+                f"Language pair {self.source_lang}-{self.target_lang} is not supported for this translator service."
+            )
+
+        import riva.client
+
+        auth = riva.client.Auth(
+            None,
+            self.use_ssl,
+            self.uri,
+            [
+                ("function-id", self.function_id),
+                ("authorization", "Bearer " + self.api_key),
+            ],
+        )
+        self.client = riva.client.NeuralMachineTranslationClient(auth)
+
+    # TODO: consider adding a backoff here and determining if a connection needs to be re-established
+    def _translate(self, text: str) -> str:
+        try:
+            response = self.client.translate(
+                [text], "", self.source_lang, self.target_lang
+            )
+            return response.translations[0].text
+        except Exception as e:
+            logging.error(f"Translation error: {str(e)}")
+            return text
+
+
+class DeeplTranslator(Translator):
+    """Remote translation using DeepL translation API
+
+    https://www.deepl.com/en/translator
+    """
+
+    ENV_VAR = "DEEPL_API_KEY"
+    DEFAULT_PARAMS = {}
+
+    # fmt: off
+    # Reference: https://developers.deepl.com/docs/resources/supported-languages
+    bcp47_support = [
+        "ar", "bg", "cs", "da", "de",  
+        "en", "el", "es", "et", "fi",
+        "fr", "hu", "id", "it", "ja",
+        "ko", "lt", "lv", "nb", "nl",
+        "pl", "pt", "ro", "ru", "sk",
+        "sl", "sv", "tr", "uk", "zh",
+        "en"
+    ]
+    # fmt: on
+
+    def _load_translator(self):
+        from deepl import Translator
+
+        if not (
+            self.source_lang in self.bcp47_support
+            and self.target_lang in self.bcp47_support
+        ):
+            raise BadGeneratorException(
+                f"Language pair {self.source_lang}-{self.target_lang} is not supported for this translator service."
+            )
+
+        self.client = Translator(self.api_key)
+
+    def _translate(self, text: str) -> str:
+        try:
+            target_lang = "EN-US" if self.target_lang == "en" else self.target_lang
+            return self.client.translate_text(
+                text, source_lang=self.source_lang, target_lang=target_lang
+            ).text
```

Comment:
we may need to handle casing - garak is bcp47 compliant but DeepL looks like it expects uppercase two-character codes